### PR TITLE
Upgrades: record google analytics events for google-voucher.

### PR DESCRIPTION
**Pull request by retrofox**
Issue: #5978 

Records events when the user does the google voucher flow.

### Testing

Set debug with `calypso:analytics`.
```
localStorage.setItem('debug', 'calypso:analytics');
```

Make the google voucher flow and take a look to the console. You should see how the events are recorded:

image

cc @artpi @mtias @rralian @Essayoh 

Test live: https://calypso.live/?branch=add/track-events-to-google-voucher

### CODE REVIEW

**Comment by lamosty** [comment]
Nice, working for me.

selection_051

The code also looks fine.

Let's :shipit: (if nobody else has any objections)

File: client/my-sites/upgrades/checkout-thank-you/google-voucher/index.jsx [hunk 1]
```diff
@@ -40,12 +41,13 @@ class GoogleVoucherDetails extends Component {
...
-		this.onButtonClick = this.onButtonClick.bind( this );
+		this.onGenerateCode = this.onGenerateCode.bind( this );
 		this.onDialogCancel = this.onDialogCancel.bind( this );
-		this.onAgreeButton = this.onAgreeButton.bind( this );
+		this.onAcceptTermsAndConditions = this.onAcceptTermsAndConditions.bind( this );
+		this.onSetupGoogleAdWordsLink = this.onSetupGoogleAdWordsLink.bind( this );
 
 		// debounced
-		this.incrementStep = debounce( this.incrementStep, 300 );
+		this.changeStep = debounce( this.changeStep, 300 );
```
Comment (lamosty) [inline client/my-sites/upgrades/checkout-thank-you/google-voucher/index.jsx]: Why it needs to be debounced?

Comment (retrofox) [inline client/my-sites/upgrades/checkout-thank-you/google-voucher/index.jsx]: Right.

**Comment by retrofox** [comment]
> I just noticed that one of the event names is not right — the last one. It should be `calypso_plans_google_voucher_setup_click but it is calypso_plans_google_voucher_setup_clic instead.

Thanks @lamosty

**Comment by artpi** [comment]
Works good:
```
calypso:analytics Recording Event ~ [Category: calypso_plans_google_voucher_generate_click] [Action: Cliked Generate Code Button] +1m

calypso:analytics Recording Event ~ [Category: calypso_plans_google_voucher_toc_accept_click] [Action: Cliked Agree Button] +3s
```

**Pull request merged**